proto/rdp/README.md

```diff
@@ -0,0 +1,277 @@
+https://github.com/reviewdog/reviewdog repository, but we may create a separate
+repository once it's reviewed and stabilized.
+
+# Reviewdog Diagnostic Protocol (RDP)
+
+Reviewdog Diagnostic Protocol (RDP. Or **R**eview**D**og **P**rotocol in short.)
+defines standard machine-readable message structures which represents a
+result of diagnostic tool such as a compiler or a linter.
+
+The idea behind the Reviewdog Diagnostic Protocol (RDP) is to standardize
+the protocol for how diagnostic tools (e.g. compilers, linters, etc..) and
+development tools (e.g. editors, reviewdog, code review API etc..) communicate.
+
+See [reviewdog.proto](reviewdog.proto) for the actual definition.
+[JSON Schema](./jsonschema) is available as well.
+
+## Wire formats of Reviewdog Diagnostic Protocol.
+
+RDP uses [Protocol Buffer](https://developers.google.com/protocol-buffers) to
+define the message structure, but recommended wire format is JSON considering
+it's widely used and easy to support both from diagnostic tools and development
+tools.
+
+### **rdpjsonl**
+JSON Lines (http://jsonlines.org/) of the [`Diagnostic`](reviewdog.proto) message ([JSON Schema](./jsonschema/Diagnostic.jsonschema)).
+
+Example:
+```json
+{"message": "<msg>", "location": {"path": "<file path>", "range": {"start": {"line": 14, "column": 15}}}, "severity": "ERROR"}
+{"message": "<msg>", "location": {"path": "<file path>", "range": {"start": {"line": 14, "column": 15}, "end": {"line": 14, "column": 18}}}, "suggestions": [{"range": {"start": {"line": 14, "column": 15}, "end": {"line": 14, "column": 18}}, "text": "<replacement text>"}], "severity": "WARNING"}
+...
+```
+
+### **rdpjson**
+JSON format of the [`DiagnosticResult`](reviewdog.proto) message ([JSON Schema](./jsonschema/DiagnosticResult.jsonschema)).
+
+Example:
+```json
+{
+  "source": {
+    "name": "super lint",
+    "url": "https://example.com/url/to/super-lint"
+  },
+  "severity": "WARNING",
+  "diagnostics": [
+    {
+      "message": "<msg>",
+      "location": {
+        "path": "<file path>",
+        "range": {
+          "start": {
+            "line": 14,
+            "column": 15
+          }
+        }
+      },
+      "severity": "ERROR",
+      "code": {
+        "value": "RULE1",
+        "url": "https://example.com/url/to/super-lint/RULE1"
+      }
+    },
+    {
+      "message": "<msg>",
+      "location": {
+        "path": "<file path>",
+        "range": {
+          "start": {
+            "line": 14,
+            "column": 15
+          },
+          "end": {
+            "line": 14,
+            "column": 18
+          }
+        }
+      },
+      "suggestions": [
+        {
+          "range": {
+            "start": {
+              "line": 14,
+              "column": 15
+            },
+            "end": {
+              "line": 14,
+              "column": 18
+            }
+          },
+          "text": "<replacement text>"
+        }
+      ],
+      "sevrity": "WARNING"
```

Comment:
Is this misspelling intentional?

(At the very least my spell checker would scream.)